molgenis-data/src/main/java/org/molgenis/data/RepositoryDecorator.java

```diff
@@ -0,0 +1,6 @@
+package org.molgenis.data;
+
+public interface RepositoryDecorator extends Repository
```

Comment:
Please add AbstractRepositoryDecorator that by default forwards all requests to the decorated repository. Add protected Repository getDecoratedRepository() that returns the decorated repository. Rename setTarget to setDecoratedRepository()